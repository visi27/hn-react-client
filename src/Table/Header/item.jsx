import React from 'react';
import classNames from 'classnames';
import { PropTypes } from 'prop-types';

import Button from '../../Button/presenter';

const HeaderItem = ({
  onSort, sortKey, style, activeSort,
}) => {
  const sortClass = classNames('button-inline', {
    'button-active': sortKey === activeSort,
  });
  return (
    <span style={style}>
      <Button className={sortClass} onClick={() => onSort(sortKey)}>
        {sortKey}
      </Button>
    </span>
  );
};

HeaderItem.propTypes = {
  onSort: PropTypes.func,
  sortKey: PropTypes.string,
  style: PropTypes.shape({
    width: PropTypes.string,
  }),
  activeSort: PropTypes.string,
};

HeaderItem.defaultProps = {
  onSort: () => {},
  sortKey: '',
  style: {
    width: '30%',
  },
  activeSort: PropTypes.string,
};

export default HeaderItem;
