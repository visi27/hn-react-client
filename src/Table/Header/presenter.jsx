import React from 'react';

const Header = () => <div className="table-header" />;
export default Header;
