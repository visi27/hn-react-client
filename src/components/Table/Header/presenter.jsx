import React from 'react';
import { PropTypes } from 'prop-types';
import * as style from '../Item/style';
import HeaderItem from './item';

const Header = ({ onSort, activeSort }) => (
  <div className="table-header">
    <HeaderItem style={style.largeColumn} onSort={onSort} sortKey="TITLE" activeSort={activeSort} />
    <HeaderItem style={style.midColumn} onSort={onSort} sortKey="AUTHOR" activeSort={activeSort} />
    <HeaderItem
      style={style.smallColumn}
      onSort={onSort}
      sortKey="COMMENTS"
      activeSort={activeSort}
    />
    <HeaderItem
      style={style.smallColumn}
      onSort={onSort}
      sortKey="POINTS"
      activeSort={activeSort}
    />
    <span style={style.smallColumn}>DISMISS</span>
  </div>
);

Header.propTypes = {
  onSort: PropTypes.func,
  activeSort: PropTypes.string,
};

Header.defaultProps = {
  onSort: () => {},
  activeSort: '',
};
export default Header;
