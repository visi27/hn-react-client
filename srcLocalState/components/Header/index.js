import React from 'react';
import * as style from '../Item/style';
import HeaderItem from './item';

const Header = ({onSort, activeSort}) => (
  <div className="table-header">
    <HeaderItem style={style.largeColumn} onSort={onSort} sortKey={'TITLE'} activeSort={activeSort}/>
    <HeaderItem style={style.midColumn} onSort={onSort} sortKey={'AUTHOR'} activeSort={activeSort}/>
    <HeaderItem style={style.smallColumn} onSort={onSort} sortKey={'COMMENTS'} activeSort={activeSort}/>
    <HeaderItem style={style.smallColumn} onSort={onSort} sortKey={'POINTS'} activeSort={activeSort}/>
    <span style={style.smallColumn}>DISMISS</span>
  </div>
);

export default Header;