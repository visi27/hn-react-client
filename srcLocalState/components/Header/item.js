import React from 'react';
import Button from '../Button';
import classNames from 'classnames';

const HeaderItem = ({onSort, sortKey, style, activeSort}) => {
  const sortClass = classNames(
    'button-inline',
    {'button-active': sortKey === activeSort}
  );
  return (
  <span style={style}>
      <Button className={sortClass} onClick={() => onSort(sortKey)}>{sortKey}</Button>
  </span>)
};

export default HeaderItem;