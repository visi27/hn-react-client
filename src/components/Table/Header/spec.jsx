import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './presenter';

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header onSort={() => {}} activeSort="COMMENTS" />, div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Header onSort={() => {}} activeSort="COMMENTS" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
