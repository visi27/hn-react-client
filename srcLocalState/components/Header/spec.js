import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Header from './index';

describe('Header', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Header onSort={() => {}} activeSort={'COMMENTS'} />, div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Header onSort={() => {}} activeSort={'COMMENTS'} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
