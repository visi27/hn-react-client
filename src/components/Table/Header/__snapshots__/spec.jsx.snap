// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header has a valid snapshot 1`] = `
<div
  className="table-header"
/>
`;
